code/modules/client/preference/character.dm

```diff
@@ -520,7 +526,9 @@
 		h_grad_offset_y = text2num(expl[2]) || 0
 	h_grad_colour = sanitize_hexcolor(h_grad_colour)
 	h_grad_alpha = sanitize_integer(h_grad_alpha, 0, 255, initial(h_grad_alpha))
-	loadout_gear	= sanitize_json(loadout_gear)
+	loadout_gear = sanitize_json(loadout_gear)
+	custom_emotes = sanitize_json(custom_emotes)
+	custom_emotes = init_custom_emotes(custom_emotes)
```

Comment:
If youre already assigning the stuff here and inside the proc with the overrides, you should either make the proc not take arguments or use a tempvar to avoid random data being saved in a var thats used by other things.
```suggestion
	loadout_gear = sanitize_json(loadout_gear)
	custom_emotes_tmp = sanitize_json(custom_emotes_tmp)
	custom_emotes = init_custom_emotes(custom_emotes_tmp)
```

The reason for doing it this way is because this single var can take on multiple forms (both a setup and un-setup list), and thats scuffed as heck.

Reply:
that makes sense